starfyre/__main__.py

```diff
@@ -9,34 +9,87 @@
     js_store = pkg_resources.path("starfyre.js", "store.js")
     shutil.copy(str(js_store), path + "/dist/store.js")
-    
-def create_main_file(path):
-    """
-    Creates a main file in the build directory.
-    
-    This file will be used to run the project.
-     Every starfyre build will have an `__init__.py` file in the build directory.
-    And the `__init__.py` file will have a component that will render the root component. It will be named `app`.
 
-    You can have a look at the `test-application/build/__init__.py` file to see what it looks like.
+def build_routes_output(generated_routes, path):
+    write_js_file(path)
 
-    The main file is also responsible for adding the `store.js` file
-    """
     output_file_path = path + "/build/__main__.py"
-    write_js_file(path)
+    init_file_path = path + "/build/__init__.py"
+    user_routes = generated_routes[:]
+    user_routes.insert(0, 'app')
+    
+    # create empty __init__.py file
+    with open(init_file_path, "w"):
+        pass
 
     with open(output_file_path, "w") as f:
         f.write(
-"""if __name__ == '__main__':
-    pass
-    """)
+            f'''from starfyre import create_component, render_root
+from starfyre.exceptions import IndexFileConflictError
 
+import os
+import sys
+from pathlib import Path
+import importlib
+
+def generate_pages(generated_routes, path):
+    """
+    Generate HTML pages for each route in the provided list of routes.
+
+    Parameters:
+    - generated_routes (list): List of generated routes.
+    - path (str): Path to the project directory.
+
+    This function generates HTML pages for each route provided in the `generated_routes` list.
+    It imports the necessary components for each route, renders them using Starfyre,
+    and writes the rendered content to corresponding HTML files.
+    """
+
+    out_dir = Path(path + "/dist").resolve()
+    root = Path(out_dir / "..").resolve()
+
+    # get the user defined project name
+    app_name = (str(root).split('/'))[-1]
+
+    for route_name in generated_routes:
+        print(f'route name is = {{route_name}}')
+        
+        if route_name.lower() == 'app':
+            component_key = 'app'  # For the 'app' component in `build/pages/__init__.py`
+        else:
+            component_key = route_name
+        
+        if route_name == 'app':
+            module_name = f"{{Path(app_name)}}.build.pages"
+        else:
+            module_name= f"{{Path(app_name)}}.build.pages.{{route_name}}"
+        
+        try:
+            module = importlib.import_module(module_name)
+            if component_key == 'app':
+                component = getattr(module, component_key, f"{{component_key}} does not exist")
+            else:
+                component = getattr(module, f'rendered_{{component_key}}')
+            result = str(component)
+        except ModuleNotFoundError:
+            print(f"Error: Could not import module '{{module_name}}'.")
+            continue
```

Comment:
Can you please explain this to me?

Reply:
The code block above is responsible for dynamically importing the correct component based on the route name and generating the HTML content for each route using that component.

```python
if route_name.lower() == 'app':
    component_key = 'app'  # For the 'app' component in `build/pages/__init__.py`
else:
    component_key = route_name
```

First, we check if the route name is 'app', and the code checks if the route_name is equal to 'app'. If it is, it sets component_key to 'app'. This is because, in the `__init__.py` file inside the build/pages directory, there might be a component named 'app'. This is a special case.


We then get the module name. We now define all compiled Python files(IR files) into the pages directory.
after this, we try to get the output of the function which is the rendered output which happened through the compiler.

and if the module is not found we skip that step though we might want to throw an error saying that we should properly import our files.